distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -1971,26 +1730,32 @@ public void testCancelDownload() throws Exception {
                 RESOURCE_DOWNLOAD_ENDPOINT_ROOT.getUrl() + "?source=" + CSW_STUB_SOURCE_ID
                         + "&metacard=" + metacardId;
 
-        // @formatter:off
-        String downloadId = given().auth().preemptive().basic(LOCALHOST_USERNAME, LOCALHOST_PASSWORD)
-                .get(startDownloadUrl).then().log().all()
-                .extract().jsonPath().getString("downloadId");
-        // @formatter:on
+        given().auth()
+                .preemptive()
+                .basic(LOCALHOST_USERNAME, LOCALHOST_PASSWORD)
+                .get(startDownloadUrl)
+                .then()
+                .log()
+                .all();
 
-        localhostCometDClient.cancelDownload(downloadId);
+        expect("Waiting for download to start.").within(40, SECONDS)
+                .until(() -> localhostCometDClient.downloadStarted());
 
-        expect("Waiting for notifications.").within(10, SECONDS)
-                .until(() -> localhostCometDClient.getMessages(NOTIFICATIONS_CHANNEL)
-                        .size() == 1);
+        localhostCometDClient.cancelAllDownloads();
 
         /**
          * Wait for 2 activities. The first one is the download started, and the second one is the
          * download canceled.
          */
-        expect("Waiting for activities.").within(10, SECONDS)
+
+        expect("Waiting for activities.").within(50, SECONDS)
                 .until(() -> localhostCometDClient.getMessages(ACTIVITIES_CHANNEL)
                         .size() == 2);
 
+        expect("Waiting for notifications.").within(50, SECONDS)
```

Comment:
Instead of waiting for an exact number of messages in the wait conditions, which is a little brittle, we could use the `getMessagesInAscOrder()` method and wait until the last message we receive is the cancellation one for the download ID we've cancelled.